plugins/modules/docker_plugin.py

```diff
@@ -0,0 +1,343 @@
+                                        active=self.existing_plugin.settings['Env'][options_count])
+
+        return differences
+
+    def install_plugin(self):
+        if not self.existing_plugin:
+            if not self.check_mode:
+                try:
+                    self.existing_plugin = self.dclient.plugins.install(self.parameters.plugin_name, None)
+                except APIError as e:
+                    self.client.fail(text_type(e))
+
+            self.results['actions'].append("Installed plugin %s" % self.parameters.plugin_name)
+            self.results['changed'] = True
+
+    def remove_plugin(self):
+        force = self.parameters.force_remove
+        if self.existing_plugin:
+            if not self.check_mode:
+                try:
+                    self.existing_plugin.remove(force)
+                except APIError as e:
+                    self.client.fail(text_type(e))
+
+            self.results['actions'].append("Removed plugin %s" % self.parameters.plugin_name)
+            self.results['changed'] = True
+
+    def update_plugin(self):
+        if self.existing_plugin:
+            differences = self.has_different_config()
+                if not differences.empty:
+                    if not self.check_mode:
+                        try:
+                            self.existing_plugin.configure(prepare_options(self.parameters.plugin_options))
+                        except APIError as e:
+                            self.client.fail(text_type(e))
+                    self.results['actions'].append("Updated plugin %s settings" % self.parameters.plugin_name)
+                    self.results['changed'] = True
+            else:
+                self.fail("Cannot update the plugin: Plugin does not exist")
+
+    def present(self):
+        differences = DifferenceTracker()
+        if self.existing_plugin:
+            differences = self.has_different_config()
+
+        self.diff_tracker.add('exists', parameter=True, active=self.existing_plugin is not None)
+
+        if self.existing_plugin:
+            self.update_plugin()
+        else:
+            self.install_plugin()
+
+        if self.diff or self.check_mode or self.parameters.debug:
+            self.diff_tracker.merge(differences)
+
+        if not self.check_mode and not self.parameters.debug:
+            self.results.pop('actions')
+
+    def absent(self):
+        self.remove_plugin()
+
+    def enable(self):
+        timeout = self.parameters.enable_timeout
+        if self.existing_plugin:
+            if not self.existing_plugin.enabled:
+                if not self.check_mode:
+                    try:
+                        self.existing_plugin.enable(timeout)
+                    except APIError as e:
+                        self.client.fail(text_type(e))
+                self.results['actions'].append("Enabled plugin %s" % self.parameters.plugin_name)
+                self.results['changed'] = True
```

Comment:
This block also needs dedenting:
```suggestion
            self.results['actions'].append("Enabled plugin %s" % self.parameters.plugin_name)
            self.results['changed'] = True
```